### PR TITLE
Add option to disable gloo process groups

### PR DESCRIPTION
**Pull request by ananthsub**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?
Take advantage of initialization option exposed from mcore from https://github.com/NVIDIA/Megatron-LM/commit/855e9426d9dddef6fc34d3b5ffd83dea7c79b53c

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by chtruong814** [comment]
@ananthsub there will be a known test failure for Flux Import.  Assuming everything else passes, let me know and I can merge it for you.

File: nemo/lightning/_strategy_lib.py [hunk 1]
```diff
@@ -138,6 +139,7 @@ def init_model_parallel(model: Optional[nn.Module] = None) -> None:
                 order="tp-cp-ep-pp-dp" if app_state.use_tp_pp_dp_mapping else "tp-cp-ep-dp-pp",
                 num_distributed_optimizer_instances=app_state.num_distributed_optimizer_instances,
                 nccl_communicator_config_path=app_state.nccl_communicator_config_path,
+                create_gloo_process_groups=app_state.use_gloo_process_groups,
```
Comment (hemildesai) [inline nemo/lightning/_strategy_lib.py:142]: Why is it called create_gloo_process_groups here but use_gloo_process_groups everywhere elss?

Comment (ananthsub) [reply nemo/lightning/_strategy_lib.py:142]: i've followed the same naming as MLM in case users search for how megatron uses this flag: https://github.com/NVIDIA/Megatron-LM/commit/855e9426d9dddef6fc34d3b5ffd83dea7c79b53c

though I agree it's confusing that parallel state's argument is named differently

Comment (hemildesai) [reply nemo/lightning/_strategy_lib.py:142]: Ah ok, let's keep it consistent with MLM then.

**Comment by chtruong814** [comment]
The one failure is a known issue we made optional.  Will go ahead and merge this.
